python/cuml/fil/fil.pyx

```diff
@@ -39,8 +39,79 @@ from cuml.utils import get_cudf_column_ptr, get_dev_array_ptr, \
 cimport cuml.common.handle
 cimport cuml.common.cuda
 
+# from treelite cimport TreeliteModel, ModelHandle
+
 cdef extern from "treelite/c_api.h":
     ctypedef void* ModelHandle
+    cdef int TreeliteLoadXGBoostModel(const char* filename,
+                                      ModelHandle* out)
+    cdef int TreeliteLoadXGBoostModelFromMemoryBuffer(const void* buf,
+                                                      size_t len,
+                                                      ModelHandle* out);
+    cdef int TreeliteFreeModel(ModelHandle handle);
+    cdef int TreeliteQueryNumTree(ModelHandle handle, size_t* out)
+    cdef int TreeliteQueryNumFeature(ModelHandle handle, size_t* out)
+    # XXX untested
+    cdef int TreeliteLoadLightGBMModel(const char* filename,
+                                           ModelHandle* out)
+    # XXX untested
+    cdef int TreeliteLoadProtobufModel(const char* filename,
+                                           ModelHandle* out)
+
+cdef class TreeliteModel():
+    cpdef ModelHandle handle
+
+    def __cinit__(self):
+        self.handle = <ModelHandle>NULL
+
+    cdef set_handle(self, ModelHandle new_handle):
+        self.handle = new_handle
+
+    cdef ModelHandle get_handle(self):
+        return self.handle
+
+    def __cdel__(self):
+        if self.handle != NULL:
+            TreeliteFreeModel(self.handle)
+
+    @property
+    def num_trees(self):
+        assert self.handle != NULL
+        cdef size_t out
+        TreeliteQueryNumTree(self.handle, &out)
+        return out
+
+    @property
+    def num_features(self):
+        assert self.handle != NULL
+        cdef size_t out
+        TreeliteQueryNumFeature(self.handle, &out)
+        return out
+
+    @staticmethod
+    def from_filename(filename, model_type="xgboost"):
+        filename_bytes = filename.encode("UTF-8")
+        cdef ModelHandle handle
+        if model_type == "xgboost":
+            res = TreeliteLoadXGBoostModel(filename_bytes, &handle)
```

Comment:
An automatic check on all treelite calls, similar to [this one](https://github.com/rapidsai/cuml/blob/branch-0.9/cpp/src/decisiontree/decisiontree_impl.h#L32), would be useful.